decidim-censuses/app/controllers/decidim/censuses/admin/census_uploads_controller.rb

```diff
@@ -0,0 +1,45 @@
+module Decidim
+  module Censuses
+    module Admin
+      class CensusUploadsController < Decidim::Admin::ApplicationController
+
+        before_action :show_instructions, unless: :module_active?
+
+        def show
+          authorize! :show, Census
+          @status = Status.new
+        end
+
+        def create
+          authorize! :create, Census
+          if params[:file]
+            data = CsvData.new(params[:file].path)
+            Census.insert_all data.values
+            RemoveDuplicatesJob.perform_later
+            flash[:notice] = t('.success', count: data.values.count,
+                                           errors: data.errors.count)
+          end
+          redirect_to census_uploads_path
+        end
+
+        def delete_all
```

Comment:
Don't be afraid of creating as many Controllers as you need. I believe that the naming of the controller is not correct enough or the method delete_all does not belong here.

Let's imagine that you don't have prior knowledge of the feature: census/uploads_controller#delete_all

I would read: Delete all the uploads from Censuses.

Maybe a more generic controller would work just fine:

resource :census, only: [:show, :create, :destroy]

You can see the census, create a new census and destroy the current census. And there is only one possible census, this is the reason why we are using resource and not resource**s**